packages/lesswrong/lib/collections/digestPosts/collection.ts

```diff
@@ -17,4 +18,13 @@ addUniversalFields({collection: DigestPosts})
 
 ensureIndex(DigestPosts, {digestId: 1})
 
+DigestPosts.checkAccess = async (user: DbUser|null, document: DbDigestPost, context: ResolverContext|null): Promise<boolean> => {
+  if (!user || !document) return false
+  if (user.isAdmin) return true
+
+  // Currently, digests become "public" once they have an end date.
+  // In the future, we might want to check the publishedDate instead.
+  return !!(await Digests.findOne({_id: document.digestId}))?.endDate
```

Comment:
Implemented in a follow up PR: https://github.com/ForumMagnum/ForumMagnum/pull/9410

Reply:
In both cases this should probably use loaders